pkg/pac/pac.go

```diff
@@ -160,42 +137,53 @@ func createSmeeDeployment(c *clients.Clients, namespace, smeeURL string) error {
 	return nil
 }
 
-func forkProject(client *gitlab.Client, projectID, targetGroupNamespace string) (*gitlab.Project, error) {
-	var project *gitlab.Project
-	var response *gitlab.Response
+func SetupSmeeDeployment() {
 	var err error
-	maxRetries := 5
-	for i := 0; i < maxRetries; i++ {
-		randomSuffix := strconv.FormatInt(time.Now().UnixNano(), 10)[:8]
-		projectName := "openshift-pipelines-test-fork-" + randomSuffix
+	smeeDeploymentName := "gosmee-client"
+	store.PutScenarioData("smee_deployment_name", smeeDeploymentName)
 
-		forkOptions := &gitlab.ForkProjectOptions{
-			Namespace: &targetGroupNamespace,
-			Name:      &projectName,
-			Path:      &projectName,
-		}
+	smeeURL, err := getNewSmeeURL()
+	if err != nil {
+		testsuit.T.Fail(fmt.Errorf("failed to get a new Smee URL: %v", err))
+	}
+	store.PutScenarioData("SMEE_URL", smeeURL)
 
-		project, response, err = client.Projects.ForkProject(projectID, forkOptions)
-		if err != nil {
-			log.Printf("Attempt %d: Failed to fork project: %v\n", i+1, err)
-			if response != nil {
-				if body, readErr := io.ReadAll(response.Body); readErr == nil {
-					log.Printf("Response body: %s\n", string(body))
-				}
-				if response.StatusCode == http.StatusConflict {
-					log.Printf("Project name or path already exists, trying again with a new name.")
-					continue
-				}
-			}
-			return nil, err
-		}
+	if err = createSmeeDeployment(store.Clients(), store.Namespace(), smeeURL); err != nil {
+		testsuit.T.Fail(fmt.Errorf("failed to create deployment: %v", err))
+	}
+}
 
-		return project, nil
+// Initialize Gitlab Client
+func InitGitLabClient() *gitlab.Client {
+	token := store.GetScenarioData("gitlabToken")
+	client, err := gitlab.NewClient(token)
+	if err != nil {
+		testsuit.T.Fail(fmt.Errorf("failed to initialize GitLab client: %v", err))
 	}
 
-	return nil, fmt.Errorf("failed to fork project after %d attempts", maxRetries)
+	return client
+}
+
+// Specified Gitlab Project ID is forked into Group Namespace
+func forkProject(client *gitlab.Client, projectID, targetNamespace string) (*gitlab.Project, error) {
+	for i := 0; i < maxRetriesForkProject; i++ {
+		projectName := fmt.Sprintf("openshift-pipelines-test-fork-%08d", time.Now().UnixNano()%1e8)
```

Comment:
I would go with something shorter

```suggestion
		projectName := fmt.Sprintf("releasetests-fork-%08d", time.Now().UnixNano()%1e8)
```